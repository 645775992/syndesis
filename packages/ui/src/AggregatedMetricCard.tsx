import {
  AggregateStatusCount,
  AggregateStatusNotification,
  AggregateStatusNotifications,
  Card,
  Icon,
} from 'patternfly-react';
import * as React from 'react';

export interface IAggregatedMetricProps {
  title: string;
  ok: number;
  error: number;
}

export class AggregatedMetricCard extends React.PureComponent<
  IAggregatedMetricProps
> {
  public render() {
    return (
      <Card accented={true} aggregated={true} matchHeight={true}>
        <Card.Title>
          <AggregateStatusCount>
            <span data-test-aggregate-title={true}>{this.props.title}</span>
          </AggregateStatusCount>
        </Card.Title>
        <Card.Body>
          <AggregateStatusNotifications>
            <AggregateStatusNotification>
              <Icon type="pf" name="ok" />
              <span data-test-aggregate-ok-count={true}>
                {this.props.ok}
              </span>{' '}
            </AggregateStatusNotification>
            <AggregateStatusNotification>
              <Icon type="pf" name="error-circle-o" />
              <span data-test-aggregate-error-count={true}>
                {this.props.error}
              </span>
            </AggregateStatusNotification>
          </AggregateStatusNotifications>
        </Card.Body>
      </Card>
    );
  }
}
