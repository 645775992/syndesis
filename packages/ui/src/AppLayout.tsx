import { Masthead, VerticalNav } from 'patternfly-react';
import * as React from 'react';
import { PfVerticalNavItem } from './PfVerticalNavItem';

export interface ILayoutBase {
  appTitle: string;
  pictograph: string;
  typogram: string;
  appNav: any;
  verticalNav: PfVerticalNavItem[];
  logoHref: string;
}

export class AppLayout extends React.Component<ILayoutBase> {
  public render() {
    return (
      <React.Fragment>
        <VerticalNav sessionKey={'mainNav'}>
          <VerticalNav.Masthead
            iconImg={this.props.pictograph}
            titleImg={this.props.typogram}
            title={this.props.appTitle}
            href={this.props.logoHref}
          >
            <Masthead.Collapse>{this.props.appNav}</Masthead.Collapse>
          </VerticalNav.Masthead>
          {this.props.verticalNav}
        </VerticalNav>
        <div className="container-pf-nav-pf-vertical">
          {this.props.children}
        </div>
      </React.Fragment>
    );
  }
}
