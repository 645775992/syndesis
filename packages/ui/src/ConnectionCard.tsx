import { Card, EmptyState } from 'patternfly-react';
import * as React from 'react';

export interface IConnectionProps {
  name: string;
  description: string;
  icon: string;
}

export class ConnectionCard extends React.PureComponent<IConnectionProps> {
  public render() {
    return (
      <Card matchHeight={true}>
        <Card.Body>
          <EmptyState>
            <div className="blank-slate-pf-icon">
              <img src={this.props.icon} alt={this.props.name} width={46}/>
            </div>
            <EmptyState.Title>
              {this.props.name}
            </EmptyState.Title>
            <EmptyState.Info>
              {this.props.description}
            </EmptyState.Info>
          </EmptyState>
        </Card.Body>
      </Card>
    );
  }
}