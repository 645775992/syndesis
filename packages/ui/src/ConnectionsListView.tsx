import * as React from 'react';
import { Link } from 'react-router-dom';
import { IListViewToolbarProps, ListViewToolbar } from './ListViewToolbar';

export interface IConnectionsListViewProps extends IListViewToolbarProps {
  linkToConnectionCreate: string;
}

export class ConnectionsListView extends React.Component<
  IConnectionsListViewProps
> {
  public render() {
    return (
      <>
        <ListViewToolbar {...this.props}>
          <div className="form-group">
            <Link
              to={this.props.linkToConnectionCreate}
              className={'btn btn-primary'}
            >
              Create Connection
            </Link>
          </div>
        </ListViewToolbar>
        <div className={'container-fluid'}>{this.props.children}</div>
      </>
    );
  }
}
