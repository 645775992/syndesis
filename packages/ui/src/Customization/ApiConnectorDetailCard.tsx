import { Card, CardBody } from 'patternfly-react';
import * as React from 'react';
import './ApiConnectorDetailCard.css';

export interface IApiConnectorDetailCardProps {
  description?: string;
  icon?: string;
  name: string;
}

export class ApiConnectorDetailCard extends React.Component<
  IApiConnectorDetailCardProps
> {
  public render() {
    return (
      <Card className="api-connector-card">
        <CardBody>
          <div className={'api-connector-card__content'}>
            <div className="api-connector-card__icon">
              <img src={this.props.icon} alt={this.props.name} />
            </div>
            <div
              className="api-connector__title h2"
              data-testid="api-connector-card-title"
            >
              {this.props.name}
            </div>
            <p className="api-connector-card__description">
              {this.props.description}
            </p>
          </div>
        </CardBody>
      </Card>
    );
  }
}
