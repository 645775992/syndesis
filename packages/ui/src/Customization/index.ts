export * from './ApiConnectorDetail';
export * from './ApiConnectorDetailCard';
export * from './ApiConnectorInfo';
export * from './ApiConnectorReview';
export * from './ApiConnectorListItem';
export * from './ApiConnectorListSkeleton';
export * from './ApiConnectorListView';
export * from './ExtensionDetail';
export * from './ExtensionImport';
export * from './ExtensionIntegrationsTable';
export * from './ExtensionListItem';
export * from './ExtensionListSkeleton';
export * from './ExtensionListView';
export * from './ExtensionOverview';
export * from './ExtensionSupports';
