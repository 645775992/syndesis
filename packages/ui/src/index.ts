export * from './AggregatedMetricCard';
export * from './ConnectionCard';
export * from './ConnectionsGrid';
export * from './ConnectionsGridCell';
export * from './ConnectionSkeleton';
export * from './ConnectionsListView';
export * from './Dashboard';
export * from './IntegrationProgress';
export * from './IntegrationsList';
export * from './IntegrationsListItem';
export * from './IntegrationsListSkeleton';
export * from './IntegrationsListView';
export * from './IntegrationStatus';
export * from './IntegrationStatusDetail';
export * from './ListViewToolbar';
export * from './ListViewToolbarAbstractComponent';
export * from './Loader';
export * from './PfNavLink';
export * from './PfVerticalNavItem';
export * from './UnrecoverableError';
