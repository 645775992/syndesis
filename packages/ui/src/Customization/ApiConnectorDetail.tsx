import {
  Card,
  CardBody,
  CardGrid,
  CardHeading,
  CardTitle,
} from 'patternfly-react';
import * as React from 'react';
import { ApiConnectorDetailCard } from './ApiConnectorDetailCard';
import { ApiConnectorInfo } from './ApiConnectorInfo';
import { ApiConnectorReview } from './ApiConnectorReview';

export interface IApiConnectorDetailProps {
  description?: string;
  icon?: string;
  name: string;
}

export class ApiConnectorDetail extends React.Component<
  IApiConnectorDetailProps
> {
  public render() {
    return (
      <CardGrid fluid={true}>
        <CardGrid.Row>
          <CardGrid.Col xs={12} md={3}>
            <ApiConnectorDetailCard
              description={this.props.description}
              icon={this.props.icon}
              name={this.props.name}
            />
          </CardGrid.Col>
          <CardGrid.Col xs={12} md={7}>
            <Card>
              <CardHeading>
                <CardTitle>{this.props.name}</CardTitle>
              </CardHeading>
              <CardBody>
                <ApiConnectorInfo />
                <ApiConnectorReview />
              </CardBody>
            </Card>
          </CardGrid.Col>
        </CardGrid.Row>
      </CardGrid>
    );
  }
}
