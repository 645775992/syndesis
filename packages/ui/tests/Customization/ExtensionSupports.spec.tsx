import * as React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import { ExtensionSupports, IAction } from '../../src/Customization';

export default describe('ExtensionSupports', () => {
  const actions = [
    {
      name: 'Action 1',
      description: 'The description for action 1',
    } as IAction,
    {
      name: 'Action 2',
      description: 'The description for action 2',
    } as IAction,
    {
      name: 'Action 3',
      description: 'The description for action 3',
    } as IAction,
    {
      name: 'Action 4',
      description: 'The description for action 4',
    } as IAction,
  ] as IAction[];

  it('Should render correctly with actions', () => {
    const componentWithActions = (
      <ExtensionSupports extensionActions={actions} />
    );
    const { queryByText } = render(componentWithActions);
    actions.map(a => {
      expect(queryByText(a.name)).toBeDefined();
      expect(queryByText(a.description)).toBeDefined();
    });
  });

  it('Should render correctly without actions', () => {
    const comp = <ExtensionSupports extensionActions={[]} />;

    // test snapshot
    const snapshot = renderer.create(comp).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
