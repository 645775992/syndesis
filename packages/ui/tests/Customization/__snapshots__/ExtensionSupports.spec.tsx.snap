// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionSupports Should render correctly without actions 1`] = `
<div
  className="extension-group"
/>
`;
